src/Geometry/GeometryRegistrar.hh

```diff
@@ -0,0 +1,45 @@
+//------------------------------------------------------------------------------
+// GeometryRegistrar
+//
+// A singleton class for maintaining global information about the type of
```

Comment:
Might be worth considering certain things on Geometry rather than Dimension later on.  That might cut down on duplicate code between Cartesian/Cylindrical/Spherical geometry as you could specialize things like kernel normalization, integration, etc. while keeping a lot of the machinery the same between them.